platform/src/main/java/org/hillview/storage/CassandraDatabase.java

```diff
@@ -136,22 +168,45 @@ public TokenRange (String rawEntry) {
         }
 
         public String toString() {
-            return "start: " + this.start_token + ", end: " + this.end_token +
-                ", endpoints: " + this.endpoints.toString();
+            StringBuilder sb = new StringBuilder();
+            sb.append("tokenRange: " + this.tokenRange.toString());
+            sb.append(", endpoints: " + this.endpoints.toString());
+            return sb.toString();
         }
     }
 
+    private String discoverLocalEndpoint() throws Exception {
+        Path cassandraPath = Paths.get(this.info.cassandraRootDir);
+        Path cassandraYaml = Paths.get(cassandraPath.toString(), CassandraDatabase.cassandraYamlPath);
+        Yaml yamlConfig = new Yaml();
+        InputStream is = new FileInputStream(cassandraYaml.toFile());
+        Config config = yamlConfig.loadAs(is, Config.class);
+        InetAddress inet = InetAddress.getByName(config.listen_address);
+        return inet.getHostAddress().toString();
```

Comment:
I wonder whether this is ever something different that localhost.

Reply:
Yes, when I deploy Cassandra to the cluster, I need to change the `listen_address` to the node IP